src/ompl/geometric/planners/experience/src/LRR.cpp

```diff
@@ -0,0 +1,610 @@
+    if (repairPlanner_)
+        repairPlanner_->clear();
+}
+
+void ompl::geometric::LRR::setLightningDB(ompl::tools::LightningDBPtr experienceDB)
+{
+    experienceDB_ = experienceDB;
+}
+
+void ompl::geometric::LRR::setRepairPlanner(const base::PlannerPtr &planner)
+{
+    if (planner && planner->getSpaceInformation().get() != si_.get())
+        throw Exception("Repair planner instance does not match space information");
+    repairPlanner_ = planner;
+    setup_ = false;
+}
+
+void ompl::geometric::LRR::setup(void)
+{
+    Planner::setup();
+
+    // Setup repair planner (for use by the rrPlanner)
+    // Note: does not use the same pdef as the main planner in this class
+    if (!repairPlanner_)
+    {
+        // Set the repair planner
+        boost::shared_ptr<ompl::geometric::RRTConnect> repair_planner( new ompl::geometric::RRTConnect( si_ ) );
+
+        OMPL_DEBUG("No repairing planner specified. Using default: %s", repair_planner->getName().c_str() );
+        repairPlanner_ = repair_planner;
+    }
+
+    // Setup the problem definition for the repair planner
+    repairProblemDef_->setOptimizationObjective(pdef_->getOptimizationObjective()); // copy primary problem def
+
+    // Setup repair planner
+    repairPlanner_->setProblemDefinition(repairProblemDef_);
+    if (!repairPlanner_->isSetup())
+        repairPlanner_->setup();
+}
+
+void ompl::geometric::LRR::freeMemory(void)
+{
+}
+
+ompl::base::PlannerStatus ompl::geometric::LRR::solve(const base::PlannerTerminationCondition &ptc)
+{
+    bool solved = false;
+    bool approximate = false;
+    double approxdif = std::numeric_limits<double>::infinity();
+
+    // Check if the database is empty
+    if (!experienceDB_->getExperiencesCount())
+    {
+        OMPL_INFORM("Experience database is empty so unable to run LRR algorithm.");
+
+        return base::PlannerStatus::CRASH;
+    }
+
+    // Get a single start state TODO: more than one
+    const base::State *startState = pis_.nextStart();
+
+    // Get a single goal state TODO: more than one
+    base::Goal *goal   = pdef_->getGoal().get();
+
+    // Check that we have the correct type of goal
+    if (!goal)
+    {
+        OMPL_ERROR("Goal cannot be converted into a goal state");
+        return base::PlannerStatus::UNRECOGNIZED_GOAL_TYPE;
+    }
+
+    const base::State *goalState = pis_.nextGoal(ptc);
+
+    // Search for previous solution in database
+    nearestPaths_ = experienceDB_->findNearestStartGoal(nearestK_, startState, goalState);
+
+    // Check if there are any solutions
+    if (nearestPaths_.empty())
+    {
+        OMPL_INFORM("No similar path founds in nearest neighbor tree, unable to retrieve repair");
+        return base::PlannerStatus::TIMEOUT; // The planner failed to find a solution
+    }
+
+    ompl::base::PlannerDataPtr chosenPath;
+
+    // Filter top n paths to 1
+    // TODO Rather than selecting 1 best path, you could also spawn n (n<=k) threads and repair the top n paths.
```

Comment:
I believe this TODO was from an earlier comment from @isucan, not on my radar atm